homeassistant/components/cover/tahoma.py

```diff
@@ -66,13 +131,56 @@ def device_class(self):
             return 'window'
         return None
 
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the sensor."""
+        attr = {}
+        super_attr = super().state_attributes
+        if super_attr is not None:
+            attr.update(super_attr)
+
+        if self._closure is not None:
+            attr[ATTR_CLOSURE] = self._closure
+        if 'core:Memorized1PositionState' in self.tahoma_device.active_states:
+            attr[ATTR_MEM_POS] = self.tahoma_device.active_states[
+                'core:Memorized1PositionState']
+        if 'core:RSSILevelState' in self.tahoma_device.active_states:
+            attr[ATTR_RSSI_LEVEL] = self.tahoma_device.active_states[
+                'core:RSSILevelState']
+        if 'core:OpenClosedState' in self.tahoma_device.active_states:
+            attr[ATTR_OPEN_CLOSE] = self.tahoma_device.active_states[
+                'core:OpenClosedState']
+        if 'core:StatusState' in self.tahoma_device.active_states:
+            attr[ATTR_STATUS] = self.tahoma_device.active_states[
+                'core:StatusState']
+        if self._lock_timer is not None:
+            attr[ATTR_LOCK_TIMER] = self._lock_timer
```

Comment:
I read the docs about the states again and I don't understand, why the lock timer should go into the state machine. It's an attribute of the device for how long it is locked (before it can be used again, e.g. open cover after wind has been detected). It relates directly to lock_level and lock_originator. Could you please point me to an example or the appropriate documentation?

Reply:
You can store timestamps, so store the exact time when it can be used. Don't store relative times that change on every update.